File: src/types/subscription.ts
export interface Subscription {
  id: string;
  name: string;
  url: string;
  last_updated: number | null;
  node_count: number;
  traffic_info: TrafficInfo | null;
}

export interface TrafficInfo {
  upload: number;
  download: number;
  total: number;
  expire: number | null;
}

File: src-tauri/src/model/subscription.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Subscription {
    pub id: String,
    pub name: String,
    pub url: String,
    pub last_updated: Option<i64>,
    pub node_count: usize,
    pub traffic_info: Option<TrafficInfo>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TrafficInfo {
    pub upload: u64,
    pub download: u64,
    pub total: u64,
    pub expire: Option<i64>,
}
